### PR TITLE
Jetpack New Pricing page - Disable manage subscription button for user who isn't a plan purchaser 

### PR DESCRIPTION
**Pull request by vikneshwar**
Completes 1202796695664022-as-1202883205133640/f

#### Changes proposed in this Pull Request

* Create a hook `useItemInfo` to get item specific details like `isNotPlanOwner`
* Use the hook and `isOwned` to disable or enable *Manage Subscription* button 
* Reuse the *OwnerInfo* component to show tooltip for disabled button
* Fix styling for the tooltip icon

#### Testing instructions

- Create a JN site and connect Jetpack
- Add a secondary admin to the site and login to the new admin account.
- Do any one of these
    * Click on Jetpack Cloud live link below and add this to the end of URL `?flags=jetpack/pricing-page-rework-v1`
    * or boot up this PR 
        * Run `git fetch && git checkout add/jp-pricing-rework/fix-wpcom-styles`
        * Run `yarn start`
        * Goto `http://calypso.localhost:3000/plans/?flags=jetpack/pricing-page-rework-v1`
- Confirm that you see the new pricing UI
- Confirm that the already purchased product has *Manage Subscription* button and it is disabled 
- Confirm that there is info icon next to *Manage Subscription* and it shows tooltip
- Now login back to main account who purchased the subscription and confirm that *Manage Subscription* is not disabled

**Screenshots**
#### A different user who did not purchase the subscription

Screenshot 2022-09-03 at 2 05 17 AM

#### User who purchased the subscription 

Screenshot 2022-09-03 at 2 01 27 AM

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202883205133640

### CODE REVIEW

File: client/my-sites/plans/jetpack-plans/product-store/featured-item-card/index.tsx [hunk 1]
```diff
@@ -16,10 +17,14 @@ export const FeaturedItemCard: React.FC< FeaturedItemCardProps > = ( {
...
 	onClickPurchase,
 	siteId,
+	purchase,
 } ) => {
 	const translate = useTranslate();
 	const { displayName: title, description } = item;
 
+	const { isNotPlanOwner } = useItemInfo( { purchase } );
+	const shouldDisableButton = isOwned && isNotPlanOwner;
+
```
Comment (manzoorwanijk) [inline client/my-sites/plans/jetpack-plans/product-store/featured-item-card/index.tsx]: Let us try to keep `FeaturedItemCard` clean and avoid any computational logic here. That way we can re-use it elsewhere.

So, we can pass a prop like `isCtaDisabled` and pass it from the parent/consumer component.

File: client/my-sites/plans/jetpack-plans/product-store/hooks/use-item-info.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+		( state ) => ! ( purchase !== undefined ? getUserOwnsPurchase( state, purchase.id ) : false )
+	);
+
+	return useMemo(
+		() => ( {
+			isNotPlanOwner,
+		} ),
+		[ isNotPlanOwner ]
+	);
+};
```
Comment (manzoorwanijk) [inline client/my-sites/plans/jetpack-plans/product-store/hooks/use-item-info.ts]: This, with some minor refactor, can be added to `useStoreItemInfo` hook.

**Review by manzoorwanijk — APPROVED** [review]
This looks good now.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7514292

Thank you @vikneshwar for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7514292).

**Pull request merged**